Mathlib/Tactic/LibrarySearch.lean

```diff
@@ -41,14 +41,36 @@ private def isBlackListed (declName : Name) : MetaM Bool := do
    || isNoConfusion env declName
   <||> isRec declName <||> isMatcher declName
 
-initialize librarySearchLemmas : DeclCache (DiscrTree Name true) ←
+/--
+A "modifier" for a declaration.
+* `none` indicates the original declaration,
+* `symm` indicates that (possibly after binders) the declaration is an `=`,
+  and we want to consider the symmetric version,
+* `mp` indicates that (possibly after binders) the declaration is an `iff`,
+  and we want to consider the forward direction,
+* `mpr` similarly, but for the backward direction.
+-/
+inductive DeclMod
+| none | symm | mp | mpr
+deriving DecidableEq
+
+initialize librarySearchLemmas : DeclCache (DiscrTree (Name × DeclMod) true) ←
   DeclCache.mk "librarySearch: init cache" {} fun name constInfo lemmas => do
     if constInfo.isUnsafe then return lemmas
     if ← isBlackListed name then return lemmas
-    withNewMCtxDepth do
+    withNewMCtxDepth do withReducible do
```

Comment:
Is this `withReducible` redundant with the other ones below?

Reply:
Thanks!